ci_tools/new_commit_with_black.sh

```diff
@@ -0,0 +1,10 @@
+black .
```

Comment:
I wonder if its better to just have black run in black_formatter_check.sh and update if needed and then in this script commit back any changes that it made. Reason being if we have it run in 2 places there is scope for them to diverge wrt to parameters. So it looks something like

1) run black and update source in repo
2) if branch != master commit repo to branch
3) run tests
4) publish

Reply:
Thanks.
I've changed the approach slightly.
There was an issue with the build on the feature branch.

**Old approach**

On feature:
(1) Run black and commit directly to feature
(2) Do package tests
(3) Check that code is black compliance

The issue with this approach - step #3 would always fails as Travis pulls the commit before black ran. I think it might cache a commit from when you start the build. I searched online but could not find a solution.

**New approach (which works)**

On feature:
(1) Run black and commit directly to feature
(2) Run package tests

On master PR
(1) Run package test
(2) Check that PR is black compliant

This approach works as the master build pulls the latest commit made with black formatting.

Reply:
on  master, it might be worth running the black check first as it will be quicker and you want to fail fast

Reply:
Thanks. Done.